### PR TITLE
badfiles: continue execution to other files instead of stopping after a checker error

### PR DESCRIPTION
**Pull request by karpinski**
#2430

### CODE REVIEW

File: beetsplug/badfiles.py [hunk 1]
```diff
@@ -97,14 +102,25 @@ def check_bad(self, lib, opts, args):
...
+                    self._log.error(
+                            u"command not found: {} when validating file: {}",
+                            e.checker,
+                            e.path)
+                else:
+                    self._log.error(u"error invoking {}: {}", e.checker, e.msg)
+                continue
             if status > 0:
-                ui.print_(u"{}: checker exited withs status {}"
+                ui.print_(u"{}: checker exited with status {}"
```
Comment (sampsyo) [inline beetsplug/badfiles.py]: :flushed: Thanks for catching this!

File: beetsplug/badfiles.py [hunk 2]
```diff
@@ -97,14 +102,25 @@ def check_bad(self, lib, opts, args):
             ext = os.path.splitext(item.path)[1][1:].decode('utf8', 'ignore')
             checker = self.get_checker(ext)
             if not checker:
-                self._log.debug(u"no checker available for {}", ext)
+                self._log.error(u"no checker specified in the config for {}",
+                                ext)
```
Comment (sampsyo) [inline beetsplug/badfiles.py]: Great; this is indeed a better message.

File: beetsplug/badfiles.py [hunk 3]
```diff
@@ -97,14 +102,25 @@ def check_bad(self, lib, opts, args):
...
                 path = item.path.decode(sys.getfilesystemencoding())
-            status, errors, output = checker(path)
+            try:
+                status, errors, output = checker(path)
+            except CheckerCommandException as e:
+                if e.errno == errno.ENOENT:
+                    self._log.error(
+                            u"command not found: {} when validating file: {}",
+                            e.checker,
+                            e.path)
```
Comment (sampsyo) [inline beetsplug/badfiles.py]: I think Travis is complaining about the indentation here. The indented lines should just be one "tab" (4 spaces) indented, and the closing parenthesis should be on a line by itself (not indented).

File: beetsplug/badfiles.py [hunk 4]
```diff
@@ -30,6 +30,16 @@
 import six
 
 
+class CheckerCommandException(Exception):
+    """Raised when running a Checker failed."""
```
Comment (sampsyo) [inline beetsplug/badfiles.py]: Could you please expand the docstring just a little to explain what the fields mean (`checker`, `path`, `errno`, and `msg`)?

Comment (karpinski) [inline beetsplug/badfiles.py]: Done

File: beetsplug/badfiles.py (hunk 3 above)
Comment (karpinski) [inline beetsplug/badfiles.py]: Fixed

**Comment by sampsyo** [comment]
Thank you for taking care of this so quickly! It's all merged up. :sparkles:

**Pull request merged**